catalog/imaging/imaging-plugin-nitf/src/main/java/org/codice/alliance/plugin/nitf/NitfPostIngestPlugin.java

```diff
@@ -115,81 +124,133 @@
+    List<Metacard> metacardUpdates = new ArrayList<>();
+    List<ContentItem> contentUpdates = new ArrayList<>();
+    for (Metacard mcard : metacards) {
+      if (shouldGenerateContentItems(mcard, properties)) {
+        createContentItems(mcard, metacardUpdates, contentUpdates);
+      }
     }
 
-    return false;
-  }
+    Map<String, Serializable> reprocessProperties = new HashMap<>();
+    reprocessProperties.put(NITF_PROCESSING_KEY, true);
+
+    if (!contentUpdates.isEmpty()) {
+      UpdateStorageRequest updateStorageRequest =
+          new UpdateStorageRequestImpl(contentUpdates, reprocessProperties);
+      try {
+        catalogFramework.update(updateStorageRequest);
+      } catch (IngestException | SourceUnavailableException e) {
+        LOGGER.debug("Error storing thumbnail/overview/original", e);
+      }
+    }
 
-  private void process(List<ContentItem> contentItems) {
-    List<ContentItem> newContentItems = new ArrayList<>();
-    contentItems.forEach(contentItem -> process(contentItem, newContentItems));
-    contentItems.addAll(newContentItems);
+    if (!metacardUpdates.isEmpty()) {
+      UpdateRequest updateRequest =
+          new UpdateRequestImpl(
+              metacardUpdates
+                  .stream()
+                  .map(
+                      mcard ->
+                          new AbstractMap.SimpleEntry<Serializable, Metacard>(mcard.getId(), mcard))
+                  .collect(Collectors.toList()),
+              Core.ID,
+              reprocessProperties);
+      try {
+        catalogFramework.update(updateRequest);
+      } catch (IngestException | SourceUnavailableException e) {
+        LOGGER.debug("Error updating metacard thumbnail", e);
+      }
+    }
   }
 
-  private void process(ContentItem contentItem, List<ContentItem> contentItems) {
-    Metacard metacard = contentItem.getMetacard();
+  private boolean shouldGenerateContentItems(
+      Metacard metacard, Map<String, Serializable> properties) {
+    Attribute type = metacard.getAttribute(Media.TYPE);
+    return type != null
+        && IMAGE_NITF.equals(type.getValue())
+        && !(boolean) properties.getOrDefault(NITF_PROCESSING_KEY, false);
+  }
 
-    if (!isNitfMimeType(contentItem.getMimeTypeRawData())) {
-      LOGGER.debug(
-          "skipping content item: filename={} mimeType={}",
-          contentItem.getFilename(),
-          contentItem.getMimeTypeRawData());
+  private void createContentItems(
+      Metacard metacard, List<Metacard> metacardUpdates, List<ContentItem> contentUpdates) {
+    ResourceResponse response;
+    try {
+      response = catalogFramework.getLocalResource(new ResourceRequestById(metacard.getId()));
+    } catch (ResourceNotFoundException | ResourceNotSupportedException | IOException e) {
+      LOGGER.debug("Error retrieving resource for thumbnail/overview/original creation", e);
       return;
     }
 
+    metacard.setAttribute(new AttributeImpl(Core.THUMBNAIL, (Serializable) null));
```

Comment:
Why set this to `null` here? If processing fails, do we want to overwrite the old `thumbnail` value? If so, there should probably be a unit test or at least javadoc for this situation.